tests/base/src/test/java/org/wildfly/security/auth/FileSystemSecurityRealmTest.java

```diff
@@ -552,9 +920,59 @@ public void testPartialIteratingTryWithResource() throws Exception {
         getRootPath(); // will fail on windows if iterator not closed correctly
     }
 
+    @Test(expected = RealmUnavailableException.class)
+    public void testMismatchSecretKey() throws Exception {
+        char[] actualPassword = "secretPassword".toCharArray();
+        PasswordFactory factory = PasswordFactory.getInstance(ClearPassword.ALGORITHM_CLEAR);
+        ClearPassword clearPassword = (ClearPassword) factory.generatePassword(new ClearPasswordSpec(actualPassword));
+        FileSystemSecurityRealm securityRealm = FileSystemSecurityRealm.builder()
+                .setRoot(getRootPath())
+                .setLevels(2)
+                .setSecretKey(key)
+                .build();
+        ModifiableRealmIdentity newIdentity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("plainUser"));
+        newIdentity.create();
+        newIdentity.setCredentials(Collections.singleton(new PasswordCredential(clearPassword)));
+        securityRealm = FileSystemSecurityRealm.builder()
+                .setRoot(getRootPath(false))
+                .setLevels(2)
+                .setSecretKey(SecretKeyUtil.generateSecretKey(192))
+                .build();
+        ModifiableRealmIdentity existingIdentity = securityRealm.getRealmIdentityForUpdate(new NamePrincipal("plainUser"));
+        existingIdentity.verifyEvidence(new PasswordGuessEvidence(actualPassword));
```

Comment:
@Ashpan With different key it is still possible to assess that the identity exists, since the username is base32 encoded, and only the reading of its attributes and the credential verification fails.

Reply:
I read your analysis document and the limitation of the username and why this was decided is well explained, thank you. I was thinking whether the existence of the identity can be obscured somehow in these cases, but didn't come up with anything that would help when the store is exposed. So this makes sense.